diff --git wordpress-orig/wp-admin/includes/file.php wordpress/wp-admin/includes/file.php
index b8a0d91..1f77a34 100644
--- wordpress-orig/wp-admin/includes/file.php
+++ wordpress/wp-admin/includes/file.php
@@ -861,7 +861,7 @@ function _wp_handle_upload( &$file, $overrides, $time, $action ) {
 	}
 
 	// Set correct file permissions.
-	$stat = stat( dirname( $new_file ));
+	$stat = stat( trailingslashit( dirname( $new_file ) ) );
 	$perms = $stat['mode'] & 0000666;
 	@ chmod( $new_file, $perms );
 
diff --git wordpress-orig/wp-includes/class-wp-image-editor-gd.php wordpress/wp-includes/class-wp-image-editor-gd.php
index c5050b2..af66bf1 100644
--- wordpress-orig/wp-includes/class-wp-image-editor-gd.php
+++ wordpress/wp-includes/class-wp-image-editor-gd.php
@@ -404,7 +404,7 @@ class WP_Image_Editor_GD extends WP_Image_Editor {
 		}
 
 		// Set correct file permissions
-		$stat = stat( dirname( $filename ) );
+		$stat = stat( trailingslashit( dirname( $filename ) ) );
 		$perms = $stat['mode'] & 0000666; //same permissions as parent folder, strip off the executable bits
 		@ chmod( $filename, $perms );
 
diff --git wordpress-orig/wp-includes/class-wp-image-editor-imagick.php wordpress/wp-includes/class-wp-image-editor-imagick.php
index 9f6a0f3..a61ce2e 100644
--- wordpress-orig/wp-includes/class-wp-image-editor-imagick.php
+++ wordpress/wp-includes/class-wp-image-editor-imagick.php
@@ -633,7 +633,7 @@ class WP_Image_Editor_Imagick extends WP_Image_Editor {
 		}
 
 		// Set correct file permissions
-		$stat = stat( dirname( $filename ) );
+		$stat = stat( trailingslashit ( dirname( $filename ) ) );
 		$perms = $stat['mode'] & 0000666; //same permissions as parent folder, strip off the executable bits
 		@ chmod( $filename, $perms );
 
diff --git wordpress-orig/wp-includes/functions.php wordpress/wp-includes/functions.php
index 58a0052..5c06904 100644
--- wordpress-orig/wp-includes/functions.php
+++ wordpress/wp-includes/functions.php
@@ -2207,7 +2207,7 @@ function wp_upload_bits( $name, $deprecated, $bits, $time = null ) {
 	clearstatcache();
 
 	// Set correct file permissions
-	$stat = @ stat( dirname( $new_file ) );
+	$stat = @ stat( trailingslashit (dirname( $new_file ) ) );
 	$perms = $stat['mode'] & 0007777;
 	$perms = $perms & 0000666;
 	@ chmod( $new_file, $perms );
